lib/feature/home_page/widgets/pokemon_tile.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:convert';
+import 'package:counter_async_redux/api/models/pokemon_model.dart';
+import 'package:counter_async_redux/api/models/types_model.dart';
+import 'package:counter_async_redux/feature/pokemon_details_page.dart/pokemon_details_page.dart';
+import 'package:counter_async_redux/utilities/constants.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+class PokemonTile extends StatefulWidget {
+  const PokemonTile({
+    required this.thisPokemon,
+  });
+  final Pokemon thisPokemon;
+
+  @override
+  _PokemonTileState createState() => _PokemonTileState();
+}
+
+class _PokemonTileState extends State<PokemonTile> {
+  TypeList? thisTileTypes;
+  @override
+  void initState() {
+    http.get(Uri.tryParse('${widget.thisPokemon.url}') ?? Uri()).then((value) {
```

Comment:
this whole code block returns an error when trying to scroll fast that states 

>  Unhandled Exception: setState() called after dispose()

Reply:
![Screenshot 2022-07-25 at 6 11 39 PM](https://user-images.githubusercontent.com/47196695/180753435-5123785a-1e79-4042-b7fd-9ba3aac6d385.png)

Reply:
I noticed this. Figuring out a better way to handle getting pokemon type